File: bnc/src/main/java/org/sqlbuilder/bnc/BNCModule.java
package org.sqlbuilder.bnc;

import org.sqlbuilder.common.Module;

import java.io.IOException;

public class BNCModule extends Module
{
	public static final String MODULE_ID = "bnc";

	protected BNCModule(final String conf, final Mode mode)
	{
		super(MODULE_ID, conf, mode);
	}

	@Override
	protected void run()
	{
		assert props != null;
		try
		{
			switch (mode)
			{
				case PLAIN:
					new BNCProcessor(props).run();
					break;
				case RESOLVE:
					new BNCResolvingProcessor(props).run();
					break;
				case UPDATE:
					new BNCUpdatingProcessor(props).run();
					break;
				default:
			}
		}
		catch (IOException | ClassNotFoundException e)
		{
			e.printStackTrace();
		}
	}

	public static void main(final String[] args)
	{
		int i = 0;
		Mode mode = Mode.PLAIN;
		if (args[i].startsWith("-"))
		{
			mode = Mode.read(args[i++]);
		}
		String conf = args[i];
		new BNCModule(conf, mode).run();
	}
}

File: bnc/src/main/java/org/sqlbuilder/bnc/BncModule.kt
package org.sqlbuilder.bnc

import org.sqlbuilder.common.Module
import org.sqlbuilder.common.Module.Mode.Companion.read
import java.io.IOException

class BncModule(
    conf: String,
    mode: Mode,
) : Module(MODULE_ID, conf, mode) {

    override fun run() {
        checkNotNull(props)
        try {
            when (mode) {
                Mode.PLAIN   -> BncProcessor(props).run()
                Mode.RESOLVE -> BncResolvingProcessor(props).run()
                Mode.UPDATE  -> BncUpdatingProcessor(props).run()
                else         -> {}
            }
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        }
    }

    companion object {

        const val MODULE_ID: String = "bnc"

        fun main(args: Array<String>) {
            var i = 0
            var mode = Mode.PLAIN
            if (args[i].startsWith("-")) {
                mode = read(args[i++])
            }
            val conf = args[i]
            BncModule(conf, mode).run()
        }
    }
}
